### ISSUE TITLE
No Ammy option in New Items

### ISSUE THREAD

**Issue by fotiDim** [AmmyUI/AmmyUI#65]
In a fresh Xamarin.Forms blank app in VS17 there is no option for Ammy Page.
![screenshot 6](https://cloud.githubusercontent.com/assets/2326415/24739744/c8b552fc-1a9d-11e7-90af-5163aa452c5a.png)

**Comment by ionoy** [comment]
Are you sure you have installed the extension? https://marketplace.visualstudio.com/items?itemName=ionoy.Ammy

![image](https://cloud.githubusercontent.com/assets/592624/24741399/5f66108c-1aae-11e7-903b-ca43ad1cb351.png)

This is what is see. I've uninstalled the extension and installed it again to be sure. Are you using VS2015 or VS2017 by the way?

**Comment by ionoy** [comment]
Anyway, at the moment you can also just create a new text file with extension Ammy and content type set to `None`.

**Comment by fotiDim** [comment]
Yes I am  sure. I tried both by "Package Manager Console" and by "Manage Nuget Packages" UI. I am on the latest Visual Studio 2017. In fact I updated today.

By the way you have a typo here: http://www.ammyui.com/documentation/xamarin-forms-quickstart/. Run `install Ammy.XamarinForms` should be `install-package Ammy.XamarinForms`

**Comment by ionoy** [comment]
I mean, did you install Visual Studio extension, not Ammy Nuget package?

I've tested it in VS2017 and it still works for me.

Thanks for the type btw, fixed it!

**Comment by fotiDim** [comment]
No, I haven't installed it. In fact, I haven't even read about it. Just
tried it and turned out this was the problem. Since it's needed you should
add it to the Xamarin.Forms guide. Also consider updating the video.

**Comment by ionoy** [comment]
Sorry for the confusion. I've updated Xamarin Forms quickstart page, it should be more visible now.

Thanks for feedback!

**Comment by fotiDim** [comment]
@ionoy another thing which is not clear in documentation is how to get hot reloading. I am running on the Android emulator now but when saving changes UI does not update. I see in the video that you set binding context to a viewModel file but a default project doesn't have something like this. Am I missing something?

**Comment by ionoy** [comment]
Hot reloading should work right away, regardless of your binding. I haven't tested Xamarin Forms support on VS2017 yet, so there might be some issues. 

What emulator are you using? 
Is active configuration set to "Debug"?
Can you try disabling firewall temporarily?
Is there `Ammy.Register="..."` attribute in generated XAML?

Unfortunately I'm not near my working PC right now, so tomorrow is the earliest time I will be able to actually test it.

**Comment by fotiDim** [comment]
Thank you for the quick response. Those are my settings. Disabling firewall didn't help. `Ammy.Register="..."` is there.

![screenshot 7](https://cloud.githubusercontent.com/assets/2326415/24771105/a4aab662-1b0c-11e7-9ffd-c614482df08b.png)

**Comment by ionoy** [comment]
Please try Android emulator that comes with Xamarin SDK.

**Comment by fotiDim** [comment]
I am using the default VS emulator. How can I use the one from Xamarin SDK? These is what I have.
![screenshot 9](https://cloud.githubusercontent.com/assets/2326415/24813074/f32c97a4-1bcc-11e7-8835-95e7c30b0440.png)
![screenshot 10](https://cloud.githubusercontent.com/assets/2326415/24813073/f3293d52-1bcc-11e7-9580-ab7fad0f8136.png)

At the same time UWP crashes on boot. Should I open another issue for that?

**Comment by ionoy** [comment]
Do you have the same emulators installed?

![image](https://cloud.githubusercontent.com/assets/592624/24829773/d5f77e7e-1c80-11e7-870d-c9ac2ab6d91f.png)

**Comment by ionoy** [comment]
Also Xamarin-UWP is not supported at the moment. I'm working on it.

**Comment by fotiDim** [comment]
I am missing `Visual Studio emulator for Android`. However as I see on your video you don't use it so guess this is not the problem.
I followed this guide: https://developer.xamarin.com/guides/android/getting_started/installation/accelerating_android_emulators/ to configure an emulator from scratch. HAXM is running successfully but still no luck. I can offer Team Viewer access is you want to debug.

**Comment by ionoy** [comment]
Please try installing "Google Android Emulator" if it's not installed yet. The problem is, I have completely different list of emulators in my VS2017. It's probably due to first installing Xamarin on VS2015.

**Comment by fotiDim** [comment]
Google Android emulator is there. In fact I even tried to reinstall it but
it didn't help. Can you share your emulator settings? Have you installed
HAXM?

On Sun, Apr 9, 2017, 18:41 Mihhail Maslakov <notifications@github.com>
wrote:

Please try installing "Google Android Emulator" if it's not installed yet.
The problem is, I have completely different list of emulators in my VS2017.
It's probably due to first installing Xamarin on VS2015.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/AmmyUI/AmmyUI/issues/65#issuecomment-292796765>, or mute
the thread

**Comment by ionoy** [comment]
By google Android emulator I mean exactly the one that was used in Xamarin demo video. Are you able to use it for debugging?

I'm working from my home laptop, there is no VS2017 installed, so it will have to wait until tomorrow, unfortunately...

**Comment by ionoy** [comment]
Also, I don't understand how to install those emulators that you have on your screenshot.

**Comment by fotiDim** [comment]
These are the default emulators coming with a clean VS17 installation. I even tried reinstalling VS and I got the same emulators.
I have a mac with Xamarin Studio that features an emulator named as yours. I tried replicating these settings on my PC but again no luck. 

- I don't get why a specific emulator is required. Is there a specific reason for that? 
- Did any other VS17 user report that it is working? 
- I would suggest a clean install on your side, perhaps on a VM.
